rdmd.d

```diff
@@ -172,9 +171,25 @@ int main(string[] args)
     if (bailout) return 0;
     if (dryRun) chatty = true; // dry-run implies chatty
     
-    // Configure the compiler to use based upon compile time information,
-    // available environmental arguments and what was requested.
-    compiler = discoverCompiler(requestedCompiler);
+    if (compiler == defaultCompiler)
```

Comment:
No, this will not work. `--compiler=dmd` is not a no-op, it disables looking for `dmd` in `rdmd`'s directory.

Reply:
Okay that would mean we have to go back to the extra variable like how I had it originally.

Reply:
Why not
```d
private string compiler = null;
...
getopt(..., &compiler);
if (!compiler)
    compiler = environment.get("RDMD_DMD", null);
if (!compiler)
{
    compiler = defaultCompiler;
    string compilerPath = buildPath(dirName(thisExePath()), compiler ~ binExt);
    if (Filesystem.existsAsFile(compilerPath))
        compiler = compilerPath;
}
```

Reply:
My main concern is something is using the compiler variable in a unittest build, so changing the default might change and break something that we don't know about.

Reply:
I see, I think that should be fine considering it's a private variable.